substrate/client/network/sync/src/strategy/state.rs

```diff
@@ -124,9 +124,7 @@ impl<B: BlockT> StateStrategy<B> {
 	}
 
 	// Create a new instance with a custom state sync provider.
-	// Used in tests.
```

Comment:
nit: Could we add a comment to mention that most of the times users are looking for `StateStrategy::new` and this should be used by custom strategies and advanced developers?